gnomad/assessment/summary_stats.py

```diff
@@ -95,37 +162,46 @@ def get_summary_counts(
+    index: int = 0,
 ) -> hl.Table:
     """
 	Generates a struct with summary counts across variant categories.
 
-	Summary counts:
-		- Number of variants
-		- Number of indels
-		- Number of SNVs
-		- Number of LoF variants
-		- Number of LoF variants that pass LOFTEE (including with LoF flags)
-		- Number of LoF variants that pass LOFTEE without LoF flags
-		- Number of OS (other splice) variants annotated by LOFTEE
-		- Number of LoF variants that fail LOFTEE filters
-
-	Also annotates Table's globals with total variant counts.
-
-	Before calculating summary counts, function:
-		- Filters out low confidence regions
-		- Filters to canonical transcripts
-		- Uses the most severe consequence 
-
-	Assumes that:
-		- Input HT is annotated with VEP.
-		- Multiallelic variants have been split and/or input HT contains bi-allelic variants only.
-
-	:param ht: Input Table.
-	:param freq_field: Name of field in HT containing frequency annotation (array of structs). Default is "freq".
-	:param filter_field: Name of field in HT containing variant filter information. Default is "filters".
-	:param filter_decoy: Whether to filter decoy regions. Default is False.
-	:return: Table grouped by frequency bin and aggregated across summary count categories. 
+    Summary counts:
+        - Number of variants
+        - Number of indels
+        - Number of SNVs
+        - Number of LoF variants
+        - Number of LoF variants that pass LOFTEE (including with LoF flags)
+        - Number of LoF variants that pass LOFTEE without LoF flags
+        - Number of OS (other splice) variants annotated by LOFTEE
+        - Number of LoF variants that fail LOFTEE filters
+
+    Also annotates Table's globals with total variant counts.
+
+    Before calculating summary counts, function:
+        - Filters out low confidence regions
+        - Filters to canonical transcripts
+        - Uses the most severe consequence 
+
+    Assumes that:
+        - Input HT is annotated with VEP.
+        - Multiallelic variants have been split and/or input HT contains bi-allelic variants only.
+        - freq_expr was calculated with `annotate_freq`.
+        - (Frequency index 0 from `annotate_freq` is frequency for all pops calculated on adj genotypes only.)
+
+    :param ht: Input Table.
+    :param freq_field: Name of field in HT containing frequency annotation (array of structs). Default is "freq".
+    :param filter_field: Name of field in HT containing variant filter information. Default is "filters".
+    :param filter_decoy: Whether to filter decoy regions. Default is False.
+    :param index: Which index of freq_expr to use for annotation. Default is 0. 
+    :return: Table grouped by frequency bin and aggregated across summary count categories. 
 	"""
+    logger.info("Checking if multi-allelic variants have been split...")
+    max_alleles = ht.aggregate(hl.agg.max(hl.len(ht.alleles)))
+    if max_alleles > 2:
+        ht = hl.split_multi(ht)
```

Comment:
Is `split_multi` sufficient here? Does it split VEP consequences as well? Since the code below relies on VEP transcript consequences (for at least `get_most_severe_consequence_for_summary`), would this also need to filter VEP consequences based on `a_index`?

Reply:
good point. I added a filter, but do you (or @konradjk) know if the VEP `allele_num`field is the same as `a_index`?

Reply:
Yep, definitely a good point. But `split_multi_hts` implements this so you can just switch to that

Reply:
I need the hail docs back